src/common/models/series/concrete-series.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2017-2018 Allegro.pl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { $, ApplyExpression, Datum, Expression, RefExpression } from "plywood";
+import { Unary } from "../../utils/functional/functional";
+import { Measure, MeasureDerivation } from "../measure/measure";
+import { TimeShiftEnv } from "../time-shift/time-shift-env";
+import { Series } from "./series";
+import { seriesFormatter } from "./series-format";
+
+export class ConcreteSeries {
+
+  constructor(public readonly series: Series, public readonly measure: Measure) {
+  }
+
+  public key(derivation = MeasureDerivation.CURRENT): string {
+    return this.plywoodKey(derivation);
+  }
+
+  /**
+   * @deprecated
+   */
+  public formatter(): Unary<number, string> {
```

Comment:
formatter makes sense for axis, because we generate data there, so formatValue doesn't make sense